app/javascript/lib/metrics-api.js

```diff
@@ -324,8 +324,8 @@ export function logMenuChange(event) {
 /**
  * Log front-end error (e.g. uncaught ReferenceError)
  */
-export function logError(text) {
-  const props = { text }
+export function logError(text, details = {}) {
+  const props = { text, ...details }
```

Comment:
Per other comment:

```suggestion
export function logError(text, error = {}) {
  logToSentry(error)
  const details = error?.message || error;
  const props = { text, ...error }
```

Reply:
oh okay cool yeah if we want to log all errors to Sentry as well this is way DRY-er to just add it here. For some reason I had thought we didn't necessarily want to log all of the errors we sent to Mixpanel to Sentry as well but I can't recall exactly why I thought that was the case. 

Do you anticipate any issue with over-logging and/or running up our quota by adding this more generally here @eweitz ?

Reply:
Ah, right, so we ideally only log application errors to Sentry, whereas user errors are also logged to Mixpanel.  An application error like this would be logged to Sentry, but e.g. a user error in validation would not be logged to Sentry.

`logError` is only called [in componentDidCatch in ErrorBoundary.jsx](https://github.com/broadinstitute/single_cell_portal_core/search?q=logerror), which also calls [logJSFetchErrorToSentry](https://github.com/broadinstitute/single_cell_portal_core/blob/a885ab2bf093902d98ec47837912ab66c449214c/app/javascript/lib/sentry-logging.js).  The latter call is _not throttled_, so all errors that pass through that function get logged to Mixpanel (via `logError`) and Sentry.

So I see a few issues, from trivial to significant:
1.  `logJSFetchErrorToSentry` seems like a misnomer, as it presumably gets (or could be) called for some errors that are not `fetch` errors.
2. `componentDidCatch` should probably just call something like your new `logError` function.  Routing JS errors to Mixpanel and Sentry seems best to make React-agnostic.
3. Why wasn't this particular "counting trace labels" error handled by [componentDidCatch](https://reactjs.org/docs/react-component.html#componentdidcatch)?  Even though the error happened in `plot.js`, this `plot.js` was used in a React component.  Naively, I'd expect `componentDidCatch` to catch errors thrown in lower-level code that's imported into a React component.

The notes above are loose observations and weak opinions.  You have more context here, so I defer to you!

Reply:
hmm okay after more deeply digging, it looks like there are two different error handling components, and I think this one: `useErrorMessage()` might be swallowing the error before `ErrorBoundary` - to point 3. And so for point 2 `componentDidCatch` is not being triggered with this error then so adding logging there won't fix this particular case (though might be a good option in the future).

For this particular case I think leaving it as this try/catch solution is sufficient, but noting that indeed  [SCP-4799](https://broadworkbench.atlassian.net/browse/SCP-4799) could make this superfluous which I think would be okay.